MasKit/Network/NetworkManager.swift

```diff
@@ -34,34 +30,35 @@ public class NetworkManager {
     /// - Parameters:
     ///   - url: URL to load data from.
     ///   - completionHandler: Closure where result is delivered.
-    func loadData(from url: URL, completionHandler: @escaping (NetworkResult) -> Void) {
-        session.loadData(from: url) { (data: Data?, error: Error?) in
-            let result: NetworkResult =
-                data != nil
-                ? .success(data!)
-                : .failure(error!)
-            completionHandler(result)
-        }
+    func loadData(from url: URL, completionHandler: @escaping (Data?, Error?) -> Void) {
+        session.loadData(from: url, completionHandler: completionHandler)
     }
 
     /// Loads data synchronously.
     ///
     /// - Parameter url: URL to load data from.
-    /// - Returns: Network result containing either Data or an Error.
-    func loadDataSync(from url: URL) -> NetworkResult {
-        var syncResult: NetworkResult?
-        let semaphore = DispatchSemaphore(value: 0)
+    /// - Returns: The Data of the response.
+    func loadDataSync(from url: URL) throws -> Data {
+        var data: Data?
+        var error: Error?
+        let group = DispatchGroup()
+        group.enter()
+        session.loadData(from: url) {
+            data = $0
+            error = $1
+            group.leave()
+        }
+
+        group.wait()
 
-        loadData(from: url) { asyncResult in
-            syncResult = asyncResult
-            semaphore.signal()
+        if let error = error {
+            throw error
         }
-        _ = semaphore.wait(timeout: .distantFuture)
 
-        guard let result = syncResult else {
-            return .failure(NetworkError.timeout)
+        if let data = data {
+            return data
         }
 
-        return result
+        throw MASError.noData
```

Comment:
Ending the method on an error case makes me a bit twitchy,. Could you switch this up and have the happy/golden path end the method?

```suggestion
        guard let data = data else {
            throw MASError.noData
        }
        
        return data
```

Reply:
I agree it's awkward. I went back and forth on this. The problem is that this definition would conflict with the definition of `data` on L42. I could force-unwrap instead?